### PR TITLE
Smartbar enhancements

### LINKED ISSUE #2049: Incognito mode indicator in smartbar like gboard
It would be nice to have an incognito indicator like gboard it replaces only the shared actions toggle icon if activated but the shared actions toggle is still functioning tho the icon is incognito icon,  sadly i cant screen shot it cause the incognito function on gbiard only activates in private tabs so its restricted. The indicator at the back of keyboard keys is quite annoying.

### LINKED ISSUE #2415: [BUG] Primary Clipboard Paste Not Fading Out And Not Disabling.
### Short description

After Updating To Latest Beta Release, Keyboard Has Primary Clip Paste Enabled, And Doesn't Disables After Primary Clip Clear. Even After Pasting Primary Clip Item, And Clearing Primary Clip, The Primary Clip Paste Button Doesn't Disables, And Doesn't Fades Away.

Screenshot :- 

Screenshot_2024-03-13-10-55-41-403_com android chrome_1

### Steps to reproduce

1. Copy Any Text, Or Just Open Keyboard After Latest Update.
2. Paste Or Clear Primary Paste.
3. See error.

### FlorisBoard Version

v0.4.0 beta

### Install Source

F-Droid

### Device

Redmi 7A

### Android

Android 10, MIUI 12.5

### PR DESCRIPTION
**Pull request by lm41**
Closes #2415
Closes #2049

### CODE REVIEW

**Comment by itsmartashub** [comment]
While you're working on this issue, I was wondering if the last copied item should still be 'pasteable' even after we remove all items from the Florisboard clipboard?
  
If this isn't default behavior but rather a bug, perhaps it's also related to the fact that even when we delete all items from the clipboard, the paste button still maintains that 'active paste' state because the **last copied item** hasn't been removed from some paste "storage" (idk how to named the place where it still hangs)
 
I'm sorry if this isn't the right place to ask.

File: app/src/main/kotlin/dev/patrickgold/florisboard/app/settings/keyboard/KeyboardScreen.kt [hunk 1]
```diff
@@ -95,6 +96,11 @@ fun KeyboardScreen() = FlorisScreen {
             max = 150,
             stepIncrement = 5,
         )
+        ListPreference(
+            listPref = prefs.keyboard.incognitoDisplayMode,
+            title = stringRes(R.string.pref__keyboard__incognito_indicator__label),
+            entries = IncognitoDisplayMode.listEntries()
```
Comment (patrickgold) [inline app/src/main/kotlin/dev/patrickgold/florisboard/app/settings/keyboard/KeyboardScreen.kt:102]:
```suggestion
            entries = IncognitoDisplayMode.listEntries(),
```

(nitpick)

File: app/src/main/kotlin/dev/patrickgold/florisboard/ime/text/keyboard/TextKeyData.kt [hunk 2]
```diff
@@ -62,7 +62,7 @@ class TextKeyData(
     }
 
     override fun asString(isForDisplay: Boolean): String {
-        return buildString {
+        return asString(this, isForDisplay)/*buildString {
```
Comment (patrickgold) [inline app/src/main/kotlin/dev/patrickgold/florisboard/ime/text/keyboard/TextKeyData.kt:65]: Please remove the commented code, as it is present in the internal helper function anyways

File: app/src/main/kotlin/dev/patrickgold/florisboard/ime/text/keyboard/TextKeyData.kt [hunk 3]
```diff
@@ -541,7 +541,8 @@ class AutoTextKeyData(
     }
 
     override fun asString(isForDisplay: Boolean): String {
-        return buildString {
+        return asString(this, isForDisplay)
+        /*return buildString {
```
Comment (patrickgold) [inline app/src/main/kotlin/dev/patrickgold/florisboard/ime/text/keyboard/TextKeyData.kt:545]: Please remove the commented code, as it is present in the internal helper function anyways

File: app/src/main/kotlin/dev/patrickgold/florisboard/app/AppPrefs.kt [hunk 4]
```diff
@@ -475,6 +476,10 @@ class AppPrefs : PreferenceModel("florisboard-app-prefs") {
             key = "keyboard__space_bar_switches_to_characters",
             default = true,
         )
+        val incognitoDisplayMode = enum(
+            key = "keyboard__incognito_indicator",
+            default = IncognitoDisplayMode.DISPLAY_BEHIND_KEYBOARD
```
Comment (patrickgold) [inline app/src/main/kotlin/dev/patrickgold/florisboard/app/AppPrefs.kt:481]:
```suggestion
            default = IncognitoDisplayMode.DISPLAY_BEHIND_KEYBOARD,
```

(nitpicks)

**Review by patrickgold — CHANGES_REQUESTED** [review]
Thanks a lot for the changes!

I have reviewed the code and tested out the code, it works really well. My requested changes are only nitpicks or code cleanup, once that's done this is good to merge in.

**Review by patrickgold — APPROVED** [review]

**Pull request merged**